hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java

```diff
@@ -3774,18 +3789,30 @@ private TableGroup createTableGroup(TableGroup parentTableGroup, SqmPath<?> join
 				querySpec.getFromClause().addRoot( tableGroup );
 			}
 			else {
-				// Check if we can reuse a table group join of the parent
-				final TableGroup compatibleTableGroup = parentTableGroup.findCompatibleJoinedGroup(
-						joinProducer,
-						SqlAstJoinType.INNER
-				);
+				final TableGroup leftJoinedTableGroup;
+				final SqlAstJoinType sqlAstJoinType;
+				if ( joinProducer instanceof ToOneAttributeMapping && ( (ToOneAttributeMapping) joinProducer ).hasNotFoundAction() ) {
+					leftJoinedTableGroup = parentTableGroup.findCompatibleJoinedGroup(
```

Comment:
Might be worth splitting extracting the join directly i.e. return the `TableGroupJoin` from a new method `TableGroup#findCompatibleJoin` and let `findCompatibleJoinedGroup` just do `TableGroupJoin j = findCompatibleJoin(...); return j == null ? null : j.getJoinedGroup();`. Then you  wouldn't have to search for the join again later below.